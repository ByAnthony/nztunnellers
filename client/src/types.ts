export type Tunneller = {
    id: number;
    serial: string;
    name: Name;
};

export type Name = {
    forename: string;
    surname: string;
};
