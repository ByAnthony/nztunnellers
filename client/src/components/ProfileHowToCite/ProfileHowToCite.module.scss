@use 'src/index.scss';

.howtocite {
  margin-top: index.$base-margin;

  p {
    margin: index.$m-margin 0;
    line-height: 1.25rem;

    @include index.tablet {
      line-height: 1.5rem;
    }
  }
}
