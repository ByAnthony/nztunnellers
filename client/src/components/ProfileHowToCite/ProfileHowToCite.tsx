import type { Summary } from '../../types/tunneller';
import { today } from '../../utils/utils';
import STYLES from './ProfileHowToCite.module.scss';

type props = {
  id: number;
  summary: Summary;
};

export function ProfileHowToCite({ id, summary }: props) {
  const displayBirthDeathDates = (birth: string, death: string | null) => (death ? `${birth}-${death}` : `${birth}-â€ ?`);
  return (
    <div className={STYLES.howtocite}>
      <h3>How to cite this page</h3>
      <p>
        {`Anthony Byledbal, "${summary.name.forename} ${summary.name.surname} `}
        {`(${displayBirthDeathDates(summary.birth, summary.death)})", `}
        <em>New Zealand Tunnellers Website</em>
        {`, ${today.getFullYear()} (2009), Accessed: `}
        {`${today.toLocaleDateString('en-NZ', {
          year: 'numeric',
          month: 'long',
          day: 'numeric',
        })}. `}
        {`URL: www.nztunnellers.com/tunnellers/${id}.`}
      </p>
    </div>
  );
}
