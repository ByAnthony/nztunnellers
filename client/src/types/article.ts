export type Section = {
    title: string,
    text: string,
}

export type Image = {
    file: string,
    title?: string;
    photographer?: string,
    reference?: string,
    alt: string,
}

export type Next = {
    url: string,
    chapter: number,
    title: string,
}

export type Article = {
    id: string,
    chapter: number,
    title: string,
    section: Section[],
    image: Image[],
    next: Next,
    notes: string,
}
