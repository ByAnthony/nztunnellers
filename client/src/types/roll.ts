export type RollInfo = {
    id: number;
    serial: string;
    name: Name;
};

type Name = {
    forename: string;
    surname: string;
};
