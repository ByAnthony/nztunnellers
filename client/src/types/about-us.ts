import { Image, Section } from './article';

export type AboutUs = {
    id: string,
    title: string,
    section: Section[],
    image: Image[],
}
