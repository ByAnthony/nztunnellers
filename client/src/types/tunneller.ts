import { Details } from './roll';

type Date = {
    year: string | undefined,
    dayMonth: string | undefined,
}

type Parent = {
    name: string | undefined,
    origin: string | undefined,
}

type Parents = {
    mother: Parent | undefined,
    father: Parent | undefined,
}

type Birth = {
    date: Date,
    country: string | undefined,
}

type Origins = {
    birth: Birth,
    parents: Parents,
    inNzLength: string,
}

type ArmyExperience = {
    unit: string | undefined,
    country: string | undefined,
    conflict: string | undefined,
    duration: string | undefined,
}

type Employment = {
    occupation: string | undefined,
    employer: string | undefined,
}

type PreWayYears = {
    armyExperience: ArmyExperience[] | [],
    employment: Employment
    residence: string | undefined;
    maritalStatus: string | undefined,
    wife: string | undefined,
    religion: string | undefined,
}

type Transferred = {
    date: Date,
    postedFrom: string,
}

type Enlistment = {
    rank: string,
    date: Date,
    district: string | undefined,
    alias: string | undefined,
    transferredToTunnellers: Transferred | undefined,
}

type Training = {
    date: Date,
    location: string,
    locationType: string,
}

type EmbarkationUnit = {
    training: Training,
    detachment: string,
    section: string | undefined,
    attachedCorps: string | undefined,
}

type Transport = {
    reference: string | undefined,
    vessel: string,
    departureDate: Date
    departurePort: string | undefined,
    arrivalDate: Date | undefined,
    arrivalPort: string | undefined,
}

type DeathPlace = {
    location: string,
    town: string,
    country: string,
}

type DeathCause = {
    type: string,
    circumstances: string,
}

type Cemetery = {
    name: string,
    location: string,
    country: string,
    graveReference: string,
}

type Death = {
    date: Date,
    place: DeathPlace | undefined,
    cause: DeathCause | undefined,
    cemetery: Cemetery | undefined,
}

type Demobilization = {
    date: Date,
    country: string | undefined,
}

type EndOfService = {
    deserter: boolean,
    transferred: Transferred | undefined,
    deathWar: Death | undefined,
    transportNz: Transport | undefined,
    demobilization: Demobilization | undefined,
}

type Medal = {
    name: string,
    country: string,
    citation: string | undefined,
}

type MilitaryYears = {
    enlistment: Enlistment,
    embarkationUnit: EmbarkationUnit,
    transportUk: Transport,
    endOfService: EndOfService,
    medals: Medal[] | [],
}

type PostWarDeath = Death & {
    deathWarInjury: boolean,
}

type PostServiceYears = {
    death: PostWarDeath,
}

type NzArchives = {
    reference: string,
    url: string,
}

type Book = {
    title: string,
    town: string,
    publisher: string,
}

type NominalRoll = Book & {
    date: string,
    page: string,
    volume: string | undefined,
    roll: string | undefined,
}

type LondonGazette = {
    page: string,
    date: string,
}

type Sources = {
    nzArchives: NzArchives[],
    awwmCenotaph: string,
    nominalRoll: NominalRoll,
    londonGazette: LondonGazette[] | [],
}

type ImageArchives = {
    location: string,
    reference: string,
}

type ImageNewspaper = {
    name: string,
    date: string,
}

type ImageBookAuthor = {
    forename: string,
    surname: string,
}

type ImageBook = Book & {
    authors: ImageBookAuthor[],
    year: string,
    page: string | undefined,
}

type ImageSource = {
    aucklandLibraries: string | undefined,
    archives: ImageArchives | undefined,
    family: string | undefined,
    newspaper: ImageNewspaper | undefined,
    book: ImageBook | undefined,
}

type Image = {
    url: string,
    source: ImageSource,
}

export type Profile = Details & {
    origins: Origins,
    preWarYears: PreWayYears,
    militaryYears: MilitaryYears,
    postServiceYears: PostServiceYears,
    sources: Sources,
    image: Image | undefined,
}
