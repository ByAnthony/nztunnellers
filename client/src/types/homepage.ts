import { Articles } from './article';
import { TunnellerImage } from './tunneller';

export type Homepage = {
    tunnellers: TunnellerImage[],
    historyChapters: Articles[],
}
