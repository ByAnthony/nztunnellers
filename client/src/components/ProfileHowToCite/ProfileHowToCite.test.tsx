import { render } from '@testing-library/react';
import { Summary } from '../../types/tunneller';
import { ProfileHowToCite } from './ProfileHowToCite';

test('renders the component', () => {
  const testId = 26;
  const testSummary: Summary = {
    serial: '12345',
    name: {
      forename: 'John',
      surname: 'Doe',
    },
    birth: '1888',
    death: '1975',
  };

  const { asFragment } = render(
    <ProfileHowToCite id={testId} summary={testSummary} />,
  );

  expect(asFragment()).toMatchSnapshot();
});
