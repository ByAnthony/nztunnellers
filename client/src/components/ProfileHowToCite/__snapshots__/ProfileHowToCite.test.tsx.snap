// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="howtocite"
  >
    <h3>
      How to cite this page
    </h3>
    <p>
      Anthony Byledbal, “John Doe (1888-1975)“,
      <em>
         New Zealand Tunnellers Website
      </em>
      , 2023 (2009), Accessed: 4 May 2023. URL: www.nztunnellers.com/tunnellers/26.
    </p>
  </div>
</DocumentFragment>
`;
